src/mono/wasm/Wasm.Build.Tests/BuildTestBase.cs

```diff
@@ -630,8 +630,9 @@ public static int Main()
     public record BuildArgs(string ProjectName,
                             string Config,
                             bool AOT,
-                            string ProjectFileContents,
-                            string? ExtraBuildArgs);
+                            string Id,
```

Comment:
Maybe I missed it, but are we using the Id parameter somewhere?

Reply:
Yes, I added it because `BuildTemplateProject` needs `Id`. Reasons for adding:
- We used to treat `ProjectFileContents` as Id, this is kind of strange but take a look e.g. here 
https://github.com/dotnet/runtime/blob/84b7c2f15ceec32e66b46231614dfcb3e334c6fd/src/mono/wasi/Wasi.Build.Tests/BuildPublishTests.cs#L31
or any other place where we construct `BuildArgs`. We always pass the value of id as the 4th argument. Then, in `ExpandBuildArgs` we trash it, overwriting `ProjectFileContents`.
- I did not want to pass another argument to `BuildIcuTest` when we already had a placeholder for `Id` in `BuildArgs`. It just required cleaning up, so that `ExpandBuildArgs` won't overwrite it.